### ISSUE TITLE
Error: RangeError [ERR_OUT_OF_RANGE]

### ISSUE THREAD

**Issue by DarminZ** [cthackers/adm-zip#426]
<img width="950" alt="image" src="https://user-images.githubusercontent.com/16647418/188084551-83b0d788-056f-4c5a-8bc3-895276388fc3.png">

"adm-zip": "^0.5.9"

internal/buffer.js:72
  throw new ERR_OUT_OF_RANGE(type || 'offset',
  ^

RangeError [ERR_OUT_OF_RANGE]: The value of "offset" is out of range. It must be >= 0 and <= 22. Received 23
    at boundsError (internal/buffer.js:72:9)
    at Buffer.readUInt16LE (internal/buffer.js:229:5)
    at parseExtra (/Users/darmin/workspace/OnTheWay/jstest/test-zip/node_modules/adm-zip/zipEntry.js:158:25)
    at Object.set extra [as extra] (/Users/darmin/workspace/OnTheWay/jstest/test-zip/node_modules/adm-zip/zipEntry.js:218:13)
    at readEntries (/Users/darmin/workspace/OnTheWay/jstest/test-zip/node_modules/adm-zip/zipFile.js:55:29)
    at Object.get entries [as entries] (/Users/darmin/workspace/OnTheWay/jstest/test-zip/node_modules/adm-zip/zipFile.js:123:17)
    at Object.getEntries (/Users/darmin/workspace/OnTheWay/jstest/test-zip/node_modules/adm-zip/adm-zip.js:474:32)
    at Object.<anonymous> (/Users/darmin/workspace/OnTheWay/jstest/test-zip/index.js:12:6)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)

**Comment by DarminZ** [comment]
[source.zip](https://github.com/DarminZ/OnTheWay/raw/master/jstest/test-zip/source.zip)

**Comment by dallenbaldwin** [comment]
I also saw this happen in one of my huge processes. 

![image](https://user-images.githubusercontent.com/14098845/190734035-52e657d4-b3e2-445a-9fc1-83d7850cefae.png)

The project I'm working on wants a massive zip of a bunch of their PDFs. By a bunch, I mean 15K+ PDFs with final pre-compression amount of 2.5GB+. It's a Civil Engineering firm and all these reports correspond to work done over the course of at least a year.

Due to the size of the number, I have a feeling I've hit an overflow issue and will probably batch these reports up in groups of 5K for now.

Regardless, I assume a more customized error should be thrown back

**Comment by nullromo** [comment]
I have the same problem. Here is a minimal example using `extractAllTo`:

```typescript
// asdf.ts
import AdmZip from 'adm-zip';
const z = new AdmZip('myFile.zip');
z.extractAllTo('./temp', true);
```

And here is the output:

```none
$ npx ts-node asdf.ts
RangeError [ERR_OUT_OF_RANGE]: The value of "offset" is out of range. It must be >= 0 and <= 22. Received 23
    at new NodeError (node:internal/errors:387:5)
    at boundsError (node:internal/buffer:86:9)
    at Buffer.readUInt16LE (node:internal/buffer:243:5)
    at parseExtra (/home/kkovacs/oss_97542/node_modules/adm-zip/zipEntry.js:158:25)
    at Object.set extra [as extra] (/home/kkovacs/oss_97542/node_modules/adm-zip/zipEntry.js:218:13)
    at readEntries (/home/kkovacs/oss_97542/node_modules/adm-zip/zipFile.js:55:29)
    at Object.get entries [as entries] (/home/kkovacs/oss_97542/node_modules/adm-zip/zipFile.js:123:17)
    at Object.extractAllTo (/home/kkovacs/oss_97542/node_modules/adm-zip/adm-zip.js:604:18)
    at Object.<anonymous> (/home/kkovacs/oss_97542/asdf.ts:3:3)
    at Module._compile (node:internal/modules/cjs/loader:1120:14) {
  code: 'ERR_OUT_OF_RANGE'
}
```

I can unzip the file on my computer just fine, so I don't think it's corrupted. The zip file is about 280KB, so it shouldn't be a size issue either.

**Comment by nullromo** [comment]
For anyone else with this problem, my current workaround is to just not use this package 😢. I can achieve the desired result using this:

```typescript
import childProcess from 'child_process';
childProcess.execSync('unzip -d ./temp -o myFile.zip');
```

Would be nice to use adm-zip here, but for now I will have to just use `unzip` in a child process.

**Comment by alitoufighi** [comment]
Same issue as @dallenbaldwin with `length` property 
Do you have any updates or do you have any suggestions to give more information?

**Comment by dallenbaldwin** [comment]
> Same issue as @dallenbaldwin with `length` property 
> Do you have any updates or do you have any suggestions to give more information?

I eventually moved to a different package and refactored my code: `zip-a-folder` has worked great so far. It's dead simple and "just works"

**Comment by alitoufighi** [comment]
Thanks @dallenbaldwin, I think I have to migrate my code too.
Anybody looking for `zip-a-folder` repo:
https://github.com/maugenst/zip-a-folder

**Comment by TheRedstoneScientist** [comment]
After some investigation, I have figured out what went wrong for me. Hopefully this will help others. The issue in my case was that the "LastModifiedTime" and "CreatedTime" were the default 1979 timestamps. I used BulkFileChanger (https://www.nirsoft.net/utils/bulk_file_changer.html) to edit all my files to have a new time. This fixed it for me. 
This happened because I copied items directly from another archive without unzipping first.
I figured this out by observing the exception happen on line 256. 
![image](https://github.com/cthackers/adm-zip/assets/7861984/484cd8ae-2318-426f-a351-bc7635b22827)

**Comment by LasyIsLazy** [comment]
> After some investigation, I have figured out what went wrong for me. Hopefully this will help others. The issue in my case was that the "LastModifiedTime" and "CreatedTime" were the default 1979 timestamps. I used BulkFileChanger (https://www.nirsoft.net/utils/bulk_file_changer.html) to edit all my files to have a new time. This fixed it for me. This happened because I copied items directly from another archive without unzipping first. I figured this out by observing the exception happen on line 256. ![image](https://private-user-images.githubusercontent.com/7861984/325231163-484cd8ae-2318-426f-a351-bc7635b22827.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQxNDcwMDEsIm5iZiI6MTcxNDE0NjcwMSwicGF0aCI6Ii83ODYxOTg0LzMyNTIzMTE2My00ODRjZDhhZS0yMzE4LTQyNmYtYTM1MS1iYzc2MzViMjI4MjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjZUMTU1MTQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmJlMTAyMjEyYzA2MGVlYzYxZTlkNmI2YTA3YWI1OWQyOWMwOWIyMjNmYWRlNDVhZTYzZmM0ZWVmNTA2OTQzYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.BkUsNcqGVJw9yv425fs4XHOfKoDtmBrZQVHaT9i9TK0)

Not my case.

**Comment by 5saviahv** [comment]
It seems to work now in 0.5.12
it was fixed in PR #449
